.github/workflows/windows-docker-image.yaml

```diff
@@ -0,0 +1,102 @@
+#
+# Create base docker image and python version specific images for windows.
+# Currently uses hub.docker.com as registry.
+#
+# Requires a DOCKERHUB_TOKEN secret and username/repository to match the github
+# repository. For example: nerdvegas/rez -> User/Repository: nerdvegas
+#
+# Commit to windows-docker-image.yaml or .github/docker/** separately
+# to trigger a image build. This Image will be used in consecutive commits.
+#
+# TODO:
+# - Better way to pass the variables throughout steps
+# - The reason for using docker is that the github supplied containers have
+#   a bloated PATH environment variable that can not be changed:
+#   https://github.community/t5/GitHub-Actions/Windows-image-bloat-in-PATH-disables-cmd-prompt-usage/m-p/36004#M2414
+# - Github offers its own package registry. Pushing to is broken in the BETA
+#   https://github.community/t5/GitHub-Actions/Docker-Push-to-Package-Registry-from-Windows/m-p/36393#M2560
+
+name: Compose Windows Docker Image
+
+on:
+  push:
+    # Will only trigger building a docker image if any of these changed
+    paths:
+    - '.github/docker/rez-win-base/**'
+    - '.github/docker/rez-win-py/**'
+    - '.github/workflows/windows-docker-image.yaml'
+
+jobs:
+
+  # ----------------------------------------------------------------------------
+  # Create base image if necessary
+  #
+  # State depends on .github\docker\rez-win-base\** and this file
+  base:
+    runs-on: windows-2019
+
+    steps:
+      - name: Login to docker repository
+        run: |
+          ${gh_user} = ("${{ github.repository }}" -Split '/')[0]
+          docker login -u ${gh_user} -p ${{ secrets.DOCKERHUB_TOKEN }}
```

Comment:
I would use the `--password-stdin` option here. So something like `echo ${{ secrets.DOCKERHUB_TOKEN }} | docker login -u ${gh_user} --password-stdin`. THis will be safer (see https://docs.docker.com/engine/reference/commandline/login/#provide-a-password-using-stdin)

We could also probably use a credential helper to be even more secure (see https://docs.docker.com/engine/reference/commandline/login/#credentials-store).

Reply:
Tried this without success:
```
Write-Output ${{ secrets.DOCKERHUB_TOKEN }} -NoNewLine | docker login -u ${gh_user} --password-stdin
```
Do you happen to know where to find a PowerShell sample to do this properly?

Reply:
I am unfortunately a complete noob in Windows... So I can't help on that front. But if it's too hard, then I guess we have to give up on this...

Reply:
I think I got it. Apparently you just need the string on Powershell:
```
'${{ secrets.DOCKERHUB_TOKEN }}' | docker login -u ${gh_user} --password-stdin
```
However - without going through a file like the example in the docs show - how would this help security? Am I not still throwing the token at the log file?
Also I was hoping that the whole 'secrets' mechanism would do the right thing to not expose it to logs. Regardless - we got it now through stdin - so at least the message does not come up.

Reply:
The actions are supposed to take care of not dumping the value of secrets in the logs, but we never know, so better be safe than sorry.

As for security, yes the password still ends up in plain text in the daemon config. That's basically why I was pointing at the different credential stores for docker.